### PR TITLE
Search documentation

### PR DESCRIPTION
**Pull request by mparkhe**
Description of syntax for search runs filters in UI and APIs.

cc: @Zangr

### CODE REVIEW

File: docs/source/query-syntax.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  params.model = "LogisticRegression" and params.lambda = "0.001" and metrics.error <= 0.05
+
+
+Special Cases
+-------------
+
+Entity name containing special characters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When metric, parameter, or tag names contain special characters like hyphen, space, period, etc
+enclose the entity name within quotes.
```
Comment (dbczumar) [inline docs/source/query-syntax.rst]: Can we specify that these must be double quotes? Single quotes do not work.

File: docs/source/query-syntax.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Example: ``metrics.accuracy``
+
+Constant
+^^^^^^^^
+
+Search syntax requires the RHS of the expression to be a constant. Type of expected constant on
+depends on LHS.
+ - If LHS of the expression is a metric, RHS is expected to be a number (integer or float)
+ - If LHS is a parameter or tag, RHS is expected to be a string constant
```
Comment (dbczumar) [inline docs/source/query-syntax.rst]: Nit: Can we explicitly stated that the string constant must be quoted in either single or double quotes?

File: docs/source/query-syntax.rst (hunk 1 above)
Comment (stbof) [reply docs/source/query-syntax.rst]: I'm editing the file and can make this fix.

File: docs/source/query-syntax.rst (hunk 2 above)
Comment (stbof) [reply docs/source/query-syntax.rst]: I'm editing the file and can make this fix.

Comment (stbof) [reply docs/source/query-syntax.rst]: @mparkhe  None of your examples have the constants in quotes. If they must be in quotes, how can the examples work?

Comment (stbof) [reply docs/source/query-syntax.rst]: Ignore my previous comment.

File: docs/source/query-syntax.rst (hunk 1 above)
Comment (mparkhe) [reply docs/source/query-syntax.rst]: Good catch. Maybe we should support both types of quotes.

File: docs/source/query-syntax.rst (hunk 2 above)
Comment (mparkhe) [reply docs/source/query-syntax.rst]: Thanks @stbof

**Review by ghost — APPROVED** [review]
Please verify you are happy with my edits.

File: docs/source/query-syntax.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+The MLflow UI and API support searching runs within a single experiment or a group of experiments
+using a search filter API. This API is a simplified version of the SQL ``WHERE`` clause.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 3
+
+Syntax
+------------------
```
Comment (mparkhe) [inline docs/source/query-syntax.rst]: Is it ok to have thus underline longer than the sub-title string? I know sphinx does not like some stuff like this

File: docs/source/query-syntax.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+the left-hand side (LHS), a comparator, and constant on the right-hand side (RHS).
+
+Identifier
+^^^^^^^^^^
+
+Required in the LHS of a search expression. Signifies an entity to compare against. An identifier has two
+parts separated by a period: the type of the entity and the name of the entity. 
+The type of the entity is ``metrics``, ``params``, or ``tags``. For example: ``metrics.accuracy``.
+
+Entity Name Contains Special Characters
```
Comment (mparkhe) [inline docs/source/query-syntax.rst]: Nice call pulling these special cases up

File: docs/source/query-syntax.rst (hunk 3 above)
Comment (stbof) [reply docs/source/query-syntax.rst]: Longer is OK. It's shorter that breaks.

File: docs/source/query-syntax.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+^^^^^^^^
+
+The search syntax requires the RHS of the expression to be a constant. The type of the constant
+depends on LHS.
+
+- If LHS is a metric, the RHS must be an integer or float number.
+- If LHS is a parameter or tag, the RHS must be a string constant enclosed in single or double quotes.
+
+
+Example Expressions
```
Comment (mparkhe) [inline docs/source/query-syntax.rst]: Does it make sense to pull these examples up top before introducing the syntax? With those special cases, some examples are already introduced.

File: docs/source/query-syntax.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+Search for runs created using a Logistic Regression model, a learning rate (lambda) of 0.001, and
+recorded error metric under 0.05.
+
+.. code-block:: sql
+
+  params.model = "LogisticRegression" and params.lambda = "0.001" and metrics.error <= 0.05
+
+
+Special Cases
```
Comment (mparkhe) [inline docs/source/query-syntax.rst]: Maybe remove this?

Comment (stbof) [reply docs/source/query-syntax.rst]: Oops. Yes.

File: docs/source/query-syntax.rst (hunk 5 above)
Comment (stbof) [reply docs/source/query-syntax.rst]: Hmm. I see your point. Would you be OK with only leaving the identifier in each example?

**Review by mparkhe — COMMENTED** [review]
Thanks, @stbof. This reads so much better.

**Comment by stbof** [comment]
@mparkhe  As a final thought, perhaps the name of the file and the anchor should match the title. Instead of query-syntax, what about search-syntax?

**Comment by mparkhe** [comment]
> @mparkhe As a final thought, perhaps the name of the file and the anchor should match the title. Instead of query-syntax, what about search-syntax?

I initially called it `search.rst` -- but it conflated with `search.html` that is already created for some "search for docs" thing. I tried a few options and it rendered as an empty page. Finally, I gave up and called it what it looks like. If `search-syntax.rst` works I am 👍 👍

**Comment by mparkhe** [comment]
@stbof : This looks great. Thanks much.

**Pull request merged**